src/WebJobs.Script/Description/DotNet/DotNetCompilationServiceFactory.cs

```diff
@@ -27,23 +27,39 @@ ImmutableArray<ScriptType> ICompilationServiceFactory.SupportedScriptTypes
             }
         }
 
-        private static OptimizationLevel OptimizationLevel
+        private static bool IsLocal => string.IsNullOrEmpty(Environment.GetEnvironmentVariable(EnvironmentSettingNames.AzureWebsiteInstanceId));
```

Comment:
I should have added a comment as I knew this was going to come up :) (I was thinking the same thing)

Yes, this should be moved to the settings manager when that lands. Ling should be landing that in soon and I'll make the change then.